pyproject.toml

```diff
@@ -257,8 +257,8 @@ required-version = "0.0.289"
 
 [tool.ruff.flake8-builtins]
 
-# We use `id` in many places and almost never want to use the python builtin.
-builtins-ignorelist = ["id"]
+# Id and type are frequently helpful as local variable or parameter names.
+builtins-ignorelist = ["id", "type"]
```

Comment:
We already exclude `id`, which is a Python builtin, and `type` is in the same category of common and useful variable name. IMO we should also change the keyword arg on `report_asset_metadata` to just `type`

Reply:
Did not realize that we do this for `id`. I'm not convinced that is a good idea either. I would rather not make this policy decision coupled with this PR.